ocs_ci/ocs/ocp.py

```diff
@@ -137,7 +137,7 @@ def exec_oc_cmd(
             if os.path.exists(cluster_dir_kubeconfig):
                 oc_cmd += f"--kubeconfig {cluster_dir_kubeconfig} "
 
-        if self.namespace:
+        if self.namespace and not command.find("-n "):
```

Comment:
I think the location of namespace flag is important...

for example:
```
$ oc rsh -n openshift-storage rook-ceph-tools-6cf5ff959b-gx5zl
sh-4.4$ exit
exit
[odedviner@localhost auth]$ oc rsh rook-ceph-tools-6cf5ff959b-gx5zl -n openshift-storage
Error from server (NotFound): pods "rook-ceph-tools-6cf5ff959b-gx5zl" not found
```

Reply:
the solution is not to pass the `-n {namespace}` argument and setting up the object namespace.
setting the object namespace is enough.